crates/ruff/src/rules/perflint/rules/incorrect_dict_iterator.rs

```diff
@@ -0,0 +1,262 @@
+                return is_not_ignored_tuple_or_name(el);
+            }
+        }
+    }
+    false
+}
+
+/// W8102
+pub(crate) fn incorrect_dict_iterator(checker: &mut Checker, target: &Expr, iter: &Expr) {
+    if let Expr::Call(ast::ExprCall { func, args, .. }) = iter {
+        let Expr::Attribute(ast::ExprAttribute { attr, value, ..  }) = func.as_ref() else {
+            return;
+        };
+        if attr != "items" {
+            return;
+        }
+        if !args.is_empty() {
+            return;
+        }
+        if let Expr::Tuple(ast::ExprTuple {
+            elts,
+            range: tup_range,
+            ..
+        }) = target
+        {
+            if elts.len() != 2 {
+                return;
+            }
+
+            match &elts[0] {
+                Expr::Name(ast::ExprName { id, range, .. }) => {
+                    if id == "_" {
+                        let mut diagnostic = Diagnostic::new(
+                            IncorrectDictIterator {
+                                subset: "values".to_string(),
+                            },
+                            *range,
+                        );
+                        if let Expr::Name(ast::ExprName {
+                            range: dict_range, ..
+                        }) = value.as_ref()
+                        {
+                            if checker.patch(diagnostic.kind.rule()) {
+                                let mut fix_val: &str = "";
+                                if let Expr::Name(ast::ExprName { id, .. }) = &elts[1] {
+                                    fix_val = id.as_str();
+                                }
+                                if let Expr::Tuple(ast::ExprTuple {
+                                    range: val_range, ..
+                                }) = &elts[1]
+                                {
+                                    fix_val = checker.locator.slice(*val_range);
+                                }
+                                diagnostic.set_fix(Fix::automatic_edits(
+                                    Edit::range_replacement(
+                                        ".values".to_string(),
+                                        TextRange::new(dict_range.end(), func.range().end()),
+                                    ),
+                                    [Edit::range_replacement(fix_val.to_string(), *tup_range)],
+                                ));
+                            }
+                            checker.diagnostics.push(diagnostic);
+                            return;
+                        }
+                    }
+                }
+                Expr::Tuple(ast::ExprTuple {
+                    elts: sub_elts,
+                    range,
+                    ..
+                }) => {
+                    if sub_elts
+                        .iter()
+                        .map(is_not_ignored_tuple_or_name)
+                        .all(|x| !x)
```

Comment:
Using `is_ignored_tuple_or_name` would make this simpler as well. So, instead of negating the result here, we'll do it in the function itself.

```suggestion
                    if !sub_elts
                        .iter()
                        .all(is_ignored_tuple_or_name)
```

If you think otherwise, using `is_not_ignored_tuple_or_name` could be simplified as:
```suggestion
                    if sub_elts
                        .iter()
                        .any(is_not_ignored_tuple_or_name)
```